### ISSUE TITLE
Creating arbitrary cross sections through SPH data

### ISSUE THREAD

**Issue by joshcalcino** [dmentipl/plonk#26]
### Describe the feature request

Implementing the ability to produce slices though the SPH data across any arbitrary plane. Right now Plonk can only do this for the xy-plane, giving slices with constant z. When this is rotated, the slice stays in the xy plane, even the plane of the disc is no longer in the xy plane (but originally was).

### Describe the solution you'd like

The ability to be able to produce cross-sectional slices across any arbitrary plane, or possibly even a 3D surface.

e.g. being able to slice across a plane inclined with respect to the xy plane.

More advanced:
Being able to define a 3D surface to, for example, take into account the flaring of a disc, so that we can see how quantities change along a constant disc scale height.

**Comment by dmentipl** [comment]
Hi @joshcalcino thanks very much for raising this feature request. It's been on my radar for a while.

This has now been implemented for an arbitrary plane (not an arbitrary surface).

For example, from the data available with the documentation:

```python
import numpy as np
import plonk
from plonk import analysis

snap = plonk.load_snap('disc_00030.h5')

snap.rotate(axis=(1, 1, 0), angle=np.pi / 3)

norm = analysis.discs.normal(snap=snap)
snap.plot('density', interp='cross_section', slice_normal=norm)
```

![tmp](https://user-images.githubusercontent.com/8281189/89726190-ead59880-da5a-11ea-9c4a-11fbfc803a8e.png)

I think it's much harder to do with an arbitrary surface because you need to know the minimum distance of a particle from the surface. I'm not sure how efficiently this can be done in general.

I'm not sure if you want to keep this issue open as a placeholder for arbitrary cross-sections.

**Comment by dmentipl** [comment]
Just for comparison, here's the projection plot:

![tmp](https://user-images.githubusercontent.com/8281189/89726215-59b2f180-da5b-11ea-850f-d757151d51f3.png)

**Comment by dmentipl** [comment]
This is available in v0.6.1.

**Comment by joshcalcino** [comment]
Looks beautiful! Great work! I think something more advanced should be a separate request.
